cmd/all-in-one/main.go

```diff
@@ -286,18 +283,17 @@ func startGRPCServer(
 
 func startZipkinHTTPAPI(
 	logger *zap.Logger,
-	zipkinPort int,
+	zipkinAddr string,
 	zipkinSpansHandler collectorApp.ZipkinSpansHandler,
 	recoveryHandler func(http.Handler) http.Handler,
 ) {
-	if zipkinPort != 0 {
+	if zipkinAddr != ":0" {
```

Comment:
":0" actually means something else, it's a way to listen on ephemeral port.

Since this is coming from a string CLI flag that used to be integer, the default value for that flag should be a blank string.